packages/frametricks/init.lua

```diff
@@ -153,12 +153,28 @@ function package:registerCommands ()
 
   self:registerCommand("showframe", function (options, _)
     local id = options.id or SILE.typesetter.frame.id
-    if id == "all" then
-      for _, frame in pairs(SILE.frames) do
-        SILE.outputter:debugFrame(frame)
+    options.allpages = options.allpages or true
+
+    local show = function()
+      if id == "all" then
+        for _, frame in pairs(SILE.frames) do
+          SILE.outputter:debugFrame(frame)
+        end
+      else
+        SILE.outputter:debugFrame(SILE.getFrame(id))
+      end
+    end
+
+    if options.allpages and options.allpages ~= "false" then
```

Comment:
See above, but since this should already be cast to a boolean at this point the logic is much simpler: `if not options.allpages then`.